### PR TITLE
Migrate FluentD Conf files to ConfigMap to make customizing FluentD conf simpler.

### PR DESCRIPTION
**Pull request by frankreno**
Move FluentD configuration files to ConfigMap to make it easier to customize.

### CODE REVIEW

File: daemonset/nonrbac/fluentd.yaml [hunk 1]
```diff
@@ -38,6 +41,33 @@ spec:
           readOnly: true
         - name: pos-files
           mountPath: /mnt/pos/
+        - name: fluentd-sumologic-config
```
Comment (bin3377) [inline daemonset/nonrbac/fluentd.yaml]: is there any reason not just mount whole directory other than individual files?

Comment (frankreno) [reply daemonset/nonrbac/fluentd.yaml]: Current Docker image seperates config across 2 directories.  `/fluentd/etc` and `fluentd/conf.d`.  So this was done this way to preserve and have a single configmap.  Other options would be:

- 2 config maps for each directory, then just have 2 volumens and mounts.
- Change entrypoint script and keep all config in single dir

Comment (frankreno) [reply daemonset/nonrbac/fluentd.yaml]: Made changes to simplify this...

**Comment by bendrucker** [comment]
@frankreno How is this going to square with the Helm chart? Seems like this would effectively require duplicating the ConfigMap here and in the chart. I generally like the move. Would love to continue to make new sources more declarative/DRY.

**Comment by frankreno** [comment]
@bendrucker : sorry for the delay.  We still need to get the release out however, we do need to square this away in there once we cut a new release.  We still need to figure out the management to avoid 2 sets of config long term.

**Pull request merged**